config/config.go

```diff
@@ -307,12 +306,16 @@ func (cfg *Config) NewNode() (host.Host, error) {
 			Peerstore:          ps,
 		}
 
-		dialer, err := autoNatCfg.makeSwarm()
+		dialerSwarm, err := autoNatCfg.makeSwarm()
+		if err != nil {
+			h.Close()
+			return nil, err
+		}
+		dialerHost, err := bhost.NewHost(dialerSwarm, nil)
```

Comment:
are we sure there is nothing listening? We need identify only and nothing else.

Also, whats this nil we are passing?

Reply:
Identify is always added: https://github.com/libp2p/go-libp2p/blob/0f76f17417428a9c1ead92f199264c88c0b031f7/p2p/host/basic/basic_host.go#L223-L227

The `nil` is a `HostOpts`: https://github.com/libp2p/go-libp2p/blob/0f76f17417428a9c1ead92f199264c88c0b031f7/p2p/host/basic/basic_host.go#L113-L155.

Reply:
can we also neuter the peerstore?

Reply:
we should pass a noop one, which we may have to shim.

Reply:
What do you mean by "neuter"? We need _some kind of_ peerstore, as autonat is first adding the peer info, and then connecting (since the swarm doesn't allow us to dial an address directly).

Reply:
ah yes, indeed.

Can we make a variant with minimal overhead that just immediately removes _everything_ when the peer is disconnected? 
We dont need a peer manager and we dont care about caching addrs.

Reply:
Maybe that should be part of autonat? The peerstore exposes a `RemovePeer` method, so we could call that one.

Reply:
that might be the right thing to do, esp if we move the pstoremgr responsibility to the host.

Only concern is we keep addrs for some 10min, as recently connected.